### PR TITLE
Add mailgun variables request

### PR DESCRIPTION
**Pull request by vitalii-bulyzhyn**
In this branch:

- request Mailgun credentials on the installation process
- render login UI conditionally
- improve user experience

### CODE REVIEW

**Comment by PJEstrada** [comment]
Hey Vitali!

Most of the code seems good to me, the only change I would suggest here is related to using the ENV variable in the frontend. This is somethin would not want for the following reasons:

- We don't want to expose that mailgun API key to our frontend systems for security reasons.
- On production we usually have the frontend service on a separate virtual machine as the backend services and potentially with different environment variables.
- We want to centralize the settings on single place in the backend and then make then front consume those settings via an API endpoint.

So  for now, for the sake of time, I think it's easier to keep the magic link button visible let's keep it that way and just make the python endpoint return an error response that we can display telling the user that the mail API key is not set.

For example:

```
Response:
Status: 400
{
"mailgun_api_is_set": false,
"message": 'The mailgun key is not set yet, please set MAILGUN_KEY"
}
```

A longer term solution would be to create a new endpoint on the backend like:

`/api/v1/settings/check/mail-api-key`

We call this during the mounting of the login screen, and based on the response of this we show/hide the magic link button.

If you think you can implement the above solution it's good too, however you feel more comfortable, I just feel that the first one might be a bit faster to implement. But I'll leave that choice to you.

The other thing is that whatever other things you encountered in the process of setting up the keys, please add them to the mail gun docs section here: https://diffgram.readme.io/docs/open-core-full-installation-mail

Let me know if I was clear enough with my comment here and as always feel free to ask me any other questions on any details you want

File: default/methods/configs/mailgun_is_set.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from methods import routes
+from shared.settings import settings
+
+@routes.route('/api/configs/is-mailer-set')
+def mailgun_is_set():
+    is_set = settings.MAILGUN_KEY and settings.EMAIL_DOMAIN_NAME and settings != 'localhost'
+    
+    if is_set:
+        return "true"
```
Comment (PJEstrada) [inline default/methods/configs/mailgun_is_set.py:9]: I think here you could  use the `jsonify` method for flask, as we always assume that the server responses will be in a valid JSON format. so you could add something like `{"result": true}` just to follow the convetion on sending JSON as responses always.

File: default/routes_init.py [hunk 2]
```diff
@@ -125,7 +125,11 @@ def do_routes_importing():
 
     from methods.ui_schema.ui_schema import ui_schema_new_api
 
+<<<<<<< HEAD
```
Comment (PJEstrada) [inline default/routes_init.py]: I think there's a merge conflict here.

File: frontend/tests/unit/user/login.spec.js [hunk 3]
```diff
@@ -40,4 +42,68 @@ describe("login.vue", () => {
     const showedMessage = wrapper.text();
     expect(showedMessage).toMatch(message);
   });
+
```
Comment (PJEstrada) [inline frontend/tests/unit/user/login.spec.js:45]: These tests look great! :)

**Review by PJEstrada — CHANGES_REQUESTED** [review]
Hey Vitali! I think code is 99% there just a couple things I pointed out in the commend and I think we can merge this one :)

File: default/methods/configs/mailgun_is_set.py (hunk 1 above)
Comment (PJEstrada) [reply default/methods/configs/mailgun_is_set.py:9]: In this case I think you'll also have to modify a bit your frontend methods to receive the value in the new format.

Comment (vitalii-bulyzhyn) [reply default/methods/configs/mailgun_is_set.py:9]: Of course, will do that

File: install.py [hunk 4]
```diff
@@ -413,10 +416,15 @@ def populate_env(self):
             env_file += 'POSTGRES_IMAGE={}\n'.format('tianon/true')
             env_file += 'DATABASE_URL={}\n'.format(self.database_url)
 
+        if self.mailgun:
+            env_file += 'MAILGUN_KEY={}\n'.format(self.mailgun_key)
+            env_file += 'EMAIL_DOMAIN_NAME={}\n'.format(self.email_domain)
```
[the remark this answers is not in the document]
Comment (vitalii-bulyzhyn) [reply install.py:421]: Oh yeah, forgot about containers

**Comment by vitalii-bulyzhyn** [comment]
@PJEstrada just fixed everything you mentioned

**Comment by PJEstrada** [comment]
> @PJEstrada just fixed everything you mentioned

Awesome thanks Vitalii! I will wait for the e2e tests to pass and if all is good we can merge this one :)

**Comment by vitalii-bulyzhyn** [comment]
@PJEstrada I saw your commit, you are trying to press type password button, but there is no that button if mailgun is not set

**Review by PJEstrada — APPROVED** [review]

**Pull request merged**